clients/src/main/java/org/apache/kafka/clients/ConnectionState.java

```diff
@@ -16,5 +16,5 @@
  * The states of a node connection
  */
 public enum ConnectionState {
-    DISCONNECTED, CONNECTING, CONNECTED
+    DISCONNECTED, CONNECTING, CONNECTED, CHECKING_API_VERSIONS, READY
```

Comment:
If the version check fails, connection is closed. No reason to go in `DISCONNECTED` state, as their is no way broker would start supporting the version without having to terminate the connection.